include/exadg/poisson/spatial_discretization/operator.cpp

```diff
@@ -136,6 +138,20 @@ Operator<dim, n_components, Number>::distribute_dofs()
   {
     affine_constraints.clear();
 
```

Comment:
Yes, we need `AffineConstraints::reinit(const IndexSet &locally_relevant_constraints)`, because otherwise `AffineConstraints` will allocate a lookup table of size proportional to the global size of the problem accumulated over all MPI processes, which will make simulations with >100m unknowns (or something around that size) to run out of memory on any number of parallel processes. The initialization to the `locally_relevant_dofs` makes sure that the lookup table only uses this selected number of DoFs, i.e., memory consumption on each MPI process is proportional to the number of unknowns on each process.

Reply:
@kronbichler I added these lines. However, I am not sure whether this is needed if we have periodic boundaries but no hanging nodes (I assumed so)? Another question is whether we should do it always, or only if `hanging_nodes or periodic_faces`?

Reply:
Yes, always do it as soon as `AffineConstraints` gets to hold any constraint. It can be skipped if you do not insert any DoF, though.